### PR TITLE
Work with no ID server set

### PR DESCRIPTION
**Pull request by dbkr**
Functionality is broken down by commit so may be easier to review commit-by-commit

Fixes https://github.com/vector-im/riot-web/issues/5921

### CODE REVIEW

File: src/components/views/auth/RegistrationForm.js [hunk 1]
```diff
@@ -91,14 +91,25 @@ module.exports = React.createClass({
...
+
+            let desc;
+            if (haveIs) {
+                desc = _t(
+                    "If you don't specify an email address, you won't be able to reset your password. " +
+                    "Are you sure?",
+                );
+            } else {
+                desc = _t(
+                    "No Identity Server is configured so you cannot add add an email address in order to " +
```
Comment (turt2live) [inline src/components/views/auth/RegistrationForm.js]: ... unless the HS handles password resets (as is the trend now)

Comment (dbkr) [reply src/components/views/auth/RegistrationForm.js]: Yeah, but this is the thing where we don't know if the HS requires an ID server to do password reset or not :(

Comment (ara4n) [reply src/components/views/auth/RegistrationForm.js]: i think we should assume the HS does the password reset for security. If someone is running a setup where the IS has to do it, the onus is on them to ensure an IS is set.

Comment (anoadragon453) [reply src/components/views/auth/RegistrationForm.js]: > you cannot add add

**Pull request merged**